source/fundamentals/crud/read-operations/sort.txt

```diff
@@ -4,11 +4,199 @@ Sort Results
+
+Example
+```````
+
+The following example specifies a descending sort on ``rating`` field:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/sort.go
+   :language: go
+   :dedent:
+   :start-after: begin descending sort
+   :end-before: end descending sort
+   :emphasize-lines: 2-3, 5
+
+After running the preceding example, the output resembles the following:
+
+.. code-block:: none
+   :copyable: false
+
+    [_id:ObjectID("...") rating:10 type:Masala]
+    [_id:ObjectID("...") rating:8 type:Earl Grey]
+    [_id:ObjectID("...") rating:7 type:Oolong]
+    [_id:ObjectID("...") rating:5 type:Assam]
+    [_id:ObjectID("...") rating:5 type:English Breakfast]
+
+Handling Ties
+~~~~~~~~~~~~~
+
+A tie occurs when two or more documents have identical values in the
+field you are using to order your results. MongoDB does not guarantee
+order in the event of ties. 
+
+For example, in the preceding examples, we encounter a tie for the
+``rating`` in the following documents:
+
+.. code-block:: none
+   :copyable: false
+
+   [_id:ObjectID("...") rating:5 type:Assam]
+   [_id:ObjectID("...") rating:5 type:English Breakfast]
+
+To guarantee a specific order for documents that have fields with
+identical values, specify additional fields to sort by in the event of a
+tie.
+
+Example
+```````
+
+The following example specifies an ascending sort on ``rating`` field,
+then a descending sort on the ``type`` field:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/sort.go
+   :language: go
+   :dedent:
+   :start-after: begin multi sort
+   :end-before: end multi sort
+   :emphasize-lines: 2-3, 5
+
+After running the preceding example, the output resembles the following:
+
+.. code-block:: none
+   :copyable: false
+
+   [_id:ObjectID("...") rating:5 type:English Breakfast]
+   [_id:ObjectID("...") rating:5 type:Assam]
+   [_id:ObjectID("...") rating:7 type:Oolong]
+   [_id:ObjectID("...") rating:8 type:Earl Grey]
+   [_id:ObjectID("...") rating:10 type:Masala]
+
+.. tip:: Using Aggregation 
+
+   You can also specify a sort in an aggregation pipeline by including
+   the :manual:`$sort </reference/operator/aggregation/sort/>` stage.
+
+   The following example specifies the same sort from the preceding
+   example in an aggregation pipeline:
+
+   .. literalinclude:: /includes/fundamentals/code-snippets/CRUD/sort.go
+      :language: go
+      :dedent:
+      :start-after: begin aggregate sort
+      :end-before: end aggregate sort
+
+Additional Information
+----------------------
+
+For more information on how to perform read operations, see our guide on
+:doc:`retrieving data </fundamentals/crud/read-operations/retrieve>`.
+
+.. For information on how to sort using an aggregation pipeline, see our
+.. guide on :doc:`Text Search </fundamentals/crud/write-operations/text>`.
+
+API Documentation
+~~~~~~~~~~~~~~~~~
+
+For more information on any of the functions or types discussed in this
```

Comment:
To learn more about functions and types discussed in this guide, see the following API Documentation: